spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -285,34 +285,26 @@
   end
 
   describe 'PUT #update' do
-    let!(:user) { create(:user) }
-    let(:params) { { id: '1982738917269816' } }
-    # let(:real_params) { { "user"=>{"name"=>"something", "about"=>"somethi", "skill_list"=>["Analytics"], 
-    #   "level_of_availability"=>"1-2 hours a day", "profile_links"=>"s", "location"=>"someth", 
-    #   "visibility"=>"0", "pair_with_projects"=>"0", "email"=>"cat@gmail.com", 
-    #   "password"=>"", "password_confirmation"=>"", "current_password"=>""} } }
+    let!(:user) { create(:user_complete_profile) }
+    let(:params) { { user: { id: user.id, email: 'changed@gmail.com' } } }
 
     before do
       sign_in user
+      session[:return_to] = projects_path
```

Comment:
Remember we were getting that `cannot find user_url` error? The error message is useless, but it is to do with the fact that we weren't setting the session hash (ie cookie) before making the `update` request in our test. In a real scenario, someone would hit the `edit` action (which would set the cookie) and then when they submit their changes they would hit the `update` action which would use the cookie. We can simulate this by setting the cookie manually in our test before making an `update` request